demos/smartlab_online_demo/python/video_processor_serial.py

```diff
@@ -0,0 +1,177 @@
+                #         front_det_results=front_det_results,
+                #         scoring=self.scoring,
+                #         state=self.state,
+                #         frame_counter=self.frame_counter)
+
+                key = cv2.waitKey(1) & 0xFF
+                if key == ord('q'):     #press 'q' to exit
+                    break
+            else:
+                print("Finished !")
+                break
+
+    def get_video_fps(self, cap):
+        return cap.get(cv2.CAP_PROP_FPS)
+
+    def get_video_total_frames(self, cap):
+        return cap.get(cv2.CAP_PROP_FRAME_COUNT)
+
+
+def build_argparser():
+    parser = ArgumentParser(add_help=False)
+    args = parser.add_argument_group('Options')
+    args.add_argument('-h', '--help', action='help', default=SUPPRESS,
+                    help='Show this help message and exit.')
+    args.add_argument('-tv', '--topview', required=True,
+                    help='Required. Topview stream to be processed. The input must be a single image, '
+                    'a folder of images, video file or camera id.')
+    args.add_argument('-fv', '--frontview', required=True,
+                    help='Required. FrontView to be processed. The input must be a single image, '
+                    'a folder of images, video file or camera id.')
+    args.add_argument('-m_ta', '--m_topall', help='Required. Path to topview all class model.', required=True, type=str)
+    args.add_argument('-m_tm', '--m_topmove', help='Required. Path to topview moving class model.', required=True, type=str)
+    args.add_argument('-m_fa', '--m_frontall', help='Required. Path to frontview all class model.', required=True, type=str)
+    args.add_argument('-m_fm', '--m_frontmove', help='Required. Path to frontview moving class model.', required=True, type=str)
+
+    subparsers = parser.add_subparsers(help='sub-command help')
+    args_mutiview = subparsers.add_parser('multiview', help='multiview help')
+    args_mutiview.add_argument('-m_en', '--m_encoder', help='Required. Path to encoder model.', required=True, type=str)
+    args_mutiview.add_argument('-m_de', '--m_decoder', help='Required. Path to decoder model.', required=True, type=str)
+    args_mutiview.add_argument('--mode', default='multiview', help='Option. Path to decoder model.', type=str)
+    args_mstcn = subparsers.add_parser('mstcn', help='mstcm help')
+    args_mstcn.add_argument('-m_i3d', '--m_i3d', help='Required. Path to i3d model.', required=True, type=str)
+    args_mstcn.add_argument('-m_mstcn', '--m_mstcn', help='Required. Path to mstcn model.', required=True, type=str)
+    args_mstcn.add_argument('--mode', default='mstcn', help='Option. Path to decoder model.', type=str)
+
+    return parser
+
+if __name__ == "__main__":
+    args = build_argparser().parse_args()
+
+    application = Application()
+    application.video_parser(top_video_path=args.topview,
+                             front_video_path=args.frontview)
```

Comment:
You can merge `Application`'s `init()` and `video_parser()`, name it `main()` and remove the class.

Reply:
fixed